### PR TITLE
Doc: Describe some PG config options

### PR DESCRIPTION
**Pull request by dbaston**
## What does this PR do?

Adds doc for some undocumented options in Postgres driver

## What are related issues/pull requests?

https://trac.osgeo.org/gdal/ticket/3744
https://trac.osgeo.org/gdal/ticket/4138

## Tasklist

 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

File: doc/source/drivers/vector/pg.rst [hunk 1]
```diff
@@ -268,6 +268,16 @@ available:
    1.333 N, where N is the size of EWKB data. However, it might be a bit
    slower than fetching in canonical form when the client and the server
    are on the same machine, so the default is NO.
+-  :decl_configoption:`PG_USE_TEXT`. If set to "YES", geometries will be
```
[the remark this answers is not in the document]
Comment (dbaston) [reply doc/source/drivers/vector/pg.rst:271]: I'd agree. I can pull out corresponding code.

File: doc/source/drivers/vector/pg.rst [hunk 2]
```diff
@@ -268,6 +268,16 @@ available:
...
+   fetched/written using WKT instead of HEX-encoded EWKB.
+-  :decl_configoption:`OGR_PG_CURSOR_PAGE`. Set the cursor page size, or
+   number of features that are fetched from the database and held in memory
+   at a single time.
+-  :decl_configoption:`OGR_PG_RETRIEVE_FID`: If set to "YES" (the default),
```
[the remark this answers is not in the document]
Comment (dbaston) [reply doc/source/drivers/vector/pg.rst:276]: This one is mentioned in the release notes (1.8.0 !) Here's one (maybe the only?) use in the wild: https://github.com/norBIT/alkisimport/blob/1d091367cca3261c035f49de6bff353dcbd4e82a/alkis-import.sh#L31

[the remark this answers is not in the document]
Comment (dbaston) [reply doc/source/drivers/vector/pg.rst:276]: I now see it in https://gdal.org/drivers/vector/pg_advanced.html#retrieving-fid-of-newly-inserted-feature

[the remark this answers is not in the document]
Comment (rouault) [reply doc/source/drivers/vector/pg.rst:276]: It seems a bit odd that external code relies on this. I let it to you to appreciate if it needs extra publicity -:-) Other drivers such as GPKG have a behavior similar to the one of  OGR_PG_RETRIEVE_FID=YES without any configurability

**Pull request merged**